File: backend/DetailsModels/SourceDetails.cs
using backend.Models;

namespace backend.DetailsModels
{
    public class SourceDetails
    {
        public Guid? Id { get; set; }
        public Guid? PersonId { get; set; }
        public Guid? SourceCategoryId { get; set; }
        public Guid? FileReferenceId { get; set; }
        public Guid? SubCategoryId { get; set; }
        public Guid? ThumbnailId { get; set; }
        public string? DownloadUrl { get; set; }
        public string? Name { get; set; } = string.Empty;
        public string? Description { get; set; } = string.Empty;
        public DateTime? DateOfCreation { get; set; }
        public DateTime? DateOfDatabaseEntry { get; set; }
        public FileReferenceDetails? FileReference { get; set; }
        public FileReferenceDetails? Thumbnail { get; set; }
        public SourceCategoryDetails? SourceCategory { get; set; }  
        public SubCategoryDetails? SubCategory { get; set; }
        public PersonDetails? Person { get; set; }
    }
}

File: frontend/src/app/api/models/source-details.ts
/* tslint:disable */
/* eslint-disable */
import { FileReferenceDetails } from './file-reference-details';
import { PersonDetails } from './person-details';
import { SourceCategoryDetails } from './source-category-details';
import { SubCategoryDetails } from './sub-category-details';
export interface SourceDetails {
  dateOfCreation?: null | string;
  dateOfDatabaseEntry?: null | string;
  description?: null | string;
  fileDownloadUrl?: null | string;
  fileReference?: FileReferenceDetails;
  fileReferenceId?: null | string;
  id?: null | string;
  name?: null | string;
  person?: PersonDetails;
  personId?: null | string;
  sourceCategory?: SourceCategoryDetails;
  sourceCategoryId?: null | string;
  subCategory?: SubCategoryDetails;
  subCategoryId?: null | string;
  thumbnail?: FileReferenceDetails;
  thumbnailDownloadUrl?: null | string;
  thumbnailId?: null | string;
}
